bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/CommunicationManager.java

```diff
@@ -520,6 +523,20 @@ public void postCommand(ChannelUID channelUID, Command command) {
         });
     }
 
+    public void sendTimeSeries(ChannelUID channelUID, TimeSeries timeSeries) {
+        ThingUID thingUID = channelUID.getThingUID();
+        Thing thing = thingRegistry.get(thingUID);
+        handleCallFromHandler(channelUID, thing, profile -> {
+            // TODO: check which profiles need enhancements
```

Comment:
What do you mean by this comment?

Reply:
Profiles need to implement `TimeSeriesProfile` to be able to properly handle time series data. I implemented the correct behavior for the `SystemDefaultProfile`. If the general design is fine, I need to check which of the other profiles in core need an enhancement to make use of the new functionality.